File: ios/Services/YandexMusic/Sources/YandexMusicService/YandexMusicStreamingConfig.swift
import Foundation

public struct YandexMusicStreamingConfig: Sendable {
    public let downloadInfoSigningSecret: String
    public let downloadInfoClientHeader: String
    public let fileInfoSigningSecret: String
    public let fileInfoClientHeader: String

    public init(
        downloadInfoSigningSecret: String = "",
        downloadInfoClientHeader: String = "",
        fileInfoSigningSecret: String = "",
        fileInfoClientHeader: String = ""
    ) {
        self.downloadInfoSigningSecret = downloadInfoSigningSecret
        self.downloadInfoClientHeader = downloadInfoClientHeader
        self.fileInfoSigningSecret = fileInfoSigningSecret
        self.fileInfoClientHeader = fileInfoClientHeader
    }

    var isDownloadInfoEnabled: Bool {
        !downloadInfoSigningSecret.trimmingCharacters(in: .whitespacesAndNewlines).isEmpty
    }

    var isFileInfoEnabled: Bool {
        !fileInfoSigningSecret.trimmingCharacters(in: .whitespacesAndNewlines).isEmpty
    }
}

File: android/services/yandex/src/main/java/com/mplayeraudio/services/yandex/internal/network/YandexMusicStreamingConfig.kt
package com.mplayeraudio.services.yandex.internal.network

internal data class YandexMusicStreamingConfig(
    val downloadInfoSigningSecret: String,
    val downloadInfoClientHeader: String,
    val fileInfoSigningSecret: String,
    val fileInfoClientHeader: String,
) {
    val isDownloadInfoEnabled: Boolean
        get() = downloadInfoSigningSecret.isNotBlank()

    val isFileInfoEnabled: Boolean
        get() = fileInfoSigningSecret.isNotBlank()
}
